assets/queries/k8s/service_does_not_target_pod/query.rego

```diff
@@ -1,56 +1,84 @@
 package Cx
 
+import data.generic.common as common_lib
 import data.generic.k8s as k8sLib
 
 CxPolicy[result] {
 	service := input.document[i]
 	service.kind == "Service"
 	metadata := service.metadata
-	ports := service.spec.ports
-	servicePorts := ports[j]
-	label := service.spec.selector[_]
-	ret := match_label(service.spec.selector)
-	count(ret) != 0
-	not confirmPorts(ret, servicePorts)
+
+	resources := [x | x := input.document[_]; matchResource(x, service.spec.selector)]
+	count(resources) > 0
+
+	servicePorts := service.spec.ports[_]
+	not confirmPorts(resources[0], servicePorts)
```

Comment:
Hello again, @Churro! 🙂 Thank you so much for your changes and great work improving this query!


I reviewed your last changes and I think `confirmPorts` should look in all resources, instead of the first one. Let me know what you think about it, please.

The following scenario led me to think about what I suggest:
- file1.yaml
```
apiVersion: v1
kind: Service
metadata:
  name: helloworld
spec:
  type: NodePort
  selector:
    app: helloworld
  ports:
    - name: http
      nodePort: 30475
      port: 8089
      protocol: TCP
      targetPort: 8089
```
- file2.yaml
```
apiVersion: v1
kind: Pod
metadata:
  name: nginx3
  labels:
    app: helloworld
spec:
  containers:
  - name: nginx
    image: nginx
    ports:
      - containerPort: 808
```
- file3.yaml
```
apiVersion: v1
kind: Pod
metadata:
  name: nginx
  labels:
    app: helloworld
spec:
  containers:
  - name: nginx
    image: nginx
    ports:
      - containerPort: 8089

```

`resources[0]` will possibly contain the pod defined in `file2.yaml` and the query will return a result because `confirmPorts` will fail. However, there is a pod that is targeted by the service in `file3.yaml`.

Reply:
Thank you for the comprehensive review, @rafaela-soares! 😊 

The use-case you describe is perfectly possible. I have now updated the rule to check all resources and hopefully you don't mind that I just took your example and added it as another negative testcase.

Reply:
Of course not 😊 Thank you so much!